### DISCUSSION TITLE
Issue running gCode - Alarm 3 Error - "Reset while in motion. Machine position is likely lost due to sudden halt."

### DISCUSSION THREAD

**Discussion by Realipj** [winder/Universal-G-Code-Sender#2444] [Help]
Hello all! I have been dealing with this issue all week and I am trying to put this out everywhere I can in hopes that someone has seen this error before. I am still very new at this CNC thing. Let's getting the general info out of the way:

I am currently running a FoxAlien Masuter Pro 4040 with the extension bed and upgraded spindle (300W) installed. I ran my spoil board surfacing pass just fine and now suddenly I am getting this error and can't run ANYTHING without it hitting the Z Limit switch.

I have gone as far as recreating the file from scratch again a but still getting this alarm. The alarm message that I am getting is:

"**An error was detected while sending 'S18000M3': (ALARM:3) Reset while in motion. Machine position is likely lost due to sudden 
halt. Re-homing is highly recommended. Streaming has been paused.**"

So all I am trying to complete is a simple 1/16" pocket pass with a 1/4" Down-cut End Mill bit. If I hit the "Home" button in UGS, it homes just fine with no issues. It registers the limit switches on all three axis. I can go ahead and set the zeros for the job with no issues but as soon as I hit the "Run" button, the router lifts the router (Z-Axis) and hits the limit switch and locks up, which is when it throws this Alarm 3 error. 

I have done some research and majority of my results point to Lightburn but I don't own the laser for this machine. Just this 300W Spindle. 

PLEASE someone help so I don't have this $1000 paper weight. :) If you want more information, please feel free to ask and i'll post it .

**Comment by breiler** [comment]
> I hit the "Run" button, the router lifts the router (Z-Axis) and hits the limit switch and locks up, which is when it throws this Alarm 3 error.

Can you post your gcode? I am suspecting that there is a `G28` command in your gcode which is used to move the machine to a predefined reference point. If this is the case, you can define a new default reference point by moving the machine to a safe location and type `G28.1`. 

To see what position will be used you can run this command `$#`.
